astropy/io/ascii/latex.py

```diff
@@ -384,7 +384,12 @@ def start_line(self, lines):
 
     def write(self, lines):
         lines.append(self.data_start)
+        lines_length_initial = len(lines)
         core.BaseData.write(self, lines)
+        # To remove extra space and // appended which creates an extra new line
+        # in the end.
+        if len(lines) > lines_length_initial:
```

Comment:
There is probably a reason why not, but could you just do (without the `len(lines) > lines_length_initial` test):
```
lines[-1] = re.sub(r'//\s*$', '', lines[-1])
```
I guess I'm not sure about a corner case of a zero-length table or something.